extensions/notebook/src/jupyter/jupyterKernel.ts

```diff
@@ -63,6 +63,12 @@ export class JupyterKernel implements nb.IKernel {
 		return true;
 	}
 
+	public get requiresConnection(): boolean {
+		// TODO would be good to have a smarter way to do this.
+		// for now only Spark kernels need a connection
+		return !!(this.kernelImpl.name && this.kernelImpl.name.toLowerCase().indexOf('spark') > -1);
```

Comment:
can this.kernelImpl be undefined?

Reply:
No. It's passed in the constructor and heavily used without null checks elsewhere. Without it everything would break.